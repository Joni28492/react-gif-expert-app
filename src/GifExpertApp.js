import React, {useState} from 'react';
import {AddCategory} from './components/AddCategory';
import { GifGrid } from './components/GifGrid';



export const GifExpertApp = ({defaultCategories=[]}) => {

    
    
    // const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState(defaultCategories);
    
    // const handleAdd = () =>{
    // //  setCategories(['HunterxHunter', ...categories]);
    //     setCategories(cats => [...cats, 'HunterxHunter']);
    // }
    
    

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr />
            
            <ol>
                {
                    categories.map(category => 
                        <GifGrid 
                            key={category}
                            category={category} />
                    )
                }
            </ol>
        </>
    )
}

