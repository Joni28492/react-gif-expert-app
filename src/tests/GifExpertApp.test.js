import { shallow } from "enzyme"
import React from 'react';
import { GifExpertApp } from "../GifExpertApp";
import '@testing-library/jest-dom';

describe('Pruebas en el componente <GifExpertApp />', () => {
    
    test('debe mostrar el componente correctamente ', () => {
        const wrapper =shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })  
    
    test('Debe de mostrar una lista de categorias', () => {

        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper =shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
})
