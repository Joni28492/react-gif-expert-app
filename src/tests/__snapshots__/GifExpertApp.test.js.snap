// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> debe mostrar el componente correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
